lib/ansible/modules/cloud/amazon/cloudformation.py

```diff
@@ -93,6 +93,13 @@
         present, the stack does exist, and neither 'template' nor 'template_url' are specified, the previous template will be reused.
     required: false
     version_added: "2.0"
+  create_changeset:
+    description:
+      - If stack already exists create a changeset instead of directly applying changes.
```

Comment:
Good point, added.
I also worked through your other remarks, checking the changesets for an existing one was a good idea. CloudFront specifies that the changeset name should be unique, even though it happily seems to create another changeset with the same name. (Bug on their side? :))